File: app/src/main/java/org/odddev/fantlab/core/utils/FragmentUtils.java
package org.odddev.fantlab.core.utils;

import android.support.annotation.IdRes;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;

/**
 * @author kenrube
 * @since 30.08.16
 */

public class FragmentUtils {

    public static void replaceFragment(FragmentActivity activity, @IdRes int containerId,
                                       Fragment fragment, boolean backStack) {
        try {
            FragmentManager fragmentManager = activity.getSupportFragmentManager();
            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.replace(containerId, fragment, fragment.getClass().getSimpleName());
            if (backStack) {
                fragmentTransaction.addToBackStack(null);
            }
            fragmentTransaction.commit();
            fragmentManager.executePendingTransactions();
        } catch (IllegalStateException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/kotlin/org/odddev/fantlab/core/utils/FragmentUtils.kt
package org.odddev.fantlab.core.utils

import android.support.annotation.IdRes
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentActivity

/**
 * @author kenrube
 * *
 * @since 30.08.16
 */

object FragmentUtils {

	fun replaceFragment(activity: FragmentActivity, @IdRes containerId: Int,
						fragment: Fragment, backStack: Boolean) {
		try {
			val fragmentManager = activity.supportFragmentManager
			val fragmentTransaction = fragmentManager.beginTransaction()
			fragmentTransaction.replace(containerId, fragment, fragment.javaClass.simpleName)
			if (backStack) {
				fragmentTransaction.addToBackStack(null)
			}
			fragmentTransaction.commit()
			fragmentManager.executePendingTransactions()
		} catch (e: IllegalStateException) {
			e.printStackTrace()
		}
	}
}
